meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor_git.bb

```diff
@@ -1,9 +1,16 @@
 require mender-monitor.inc
 
-PV = "master-git"
-
-# Source directly the repo, override name
-S = "${WORKDIR}/monitor-client"
+def mender_monitor_version_from_preferred_version(d):
+    pref_version = d.getVar("PREFERRED_VERSION")
+    if pref_version is not None and pref_version.find("-build") >= 0:
+        # If "-build" is in the version, use the version as is. This means that
+        # we can build tags with "-build" in them from this recipe, but not
+        # final tags, which will need their own recipe.
+        return pref_version
+    else:
+        # Else return the default "master-git".
+        return "master-git"
+PV = "${@mender_monitor_version_from_preferred_version(d)}"
```

Comment:
just for my understanding, where is `d` defined here?

Reply:
It is the Yocto built-in Python dictionary with all the meta-data for each recipe. See [documentation](https://www.yoctoproject.org/docs/current/mega-manual/mega-manual.html#inline-python-variable-expansion).